provider/v2/aws/provider.go

```diff
@@ -34,14 +38,45 @@ type Provider struct {
 }
 
 func (p *Provider) Kind() apitypes.CloudProvider {
-	// TODO implement me
-	panic("implement me")
+	return apitypes.AWS
 }
 
-func New(_ context.Context) (provider.Provider, error) {
+func New(ctx context.Context) (provider.Provider, error) {
+	config, err := NewConfig()
+	if err != nil {
+		return nil, fmt.Errorf("invalid configuration. Provider=AWS: %w", err)
+	}
+
+	if err = config.Validate(); err != nil {
+		return nil, fmt.Errorf("failed to validate provider configuration. Provider=AWS: %w", err)
+	}
+
+	cfg, err := awsconfig.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load aws config: %w", err)
+	}
+
+	ec2Client := ec2.NewFromConfig(cfg)
+
 	return &Provider{
-		Discoverer: &discoverer.Discoverer{},
-		Scanner:    &scanner.Scanner{},
-		Estimator:  &estimator.Estimator{},
+		Discoverer: &discoverer.Discoverer{
+			Ec2Client: ec2Client,
+		},
+		Scanner: &scanner.Scanner{
+			Kind:                apitypes.AWS,
+			ScannerRegion:       config.ScannerRegion,
+			BlockDeviceName:     config.BlockDeviceName,
+			ScannerImage:        config.ScannerImage,
+			ScannerInstanceType: config.ScannerInstanceType,
+			SecurityGroupID:     config.SecurityGroupID,
+			SubnetID:            config.SubnetID,
+			KeyPairName:         config.KeyPairName,
+			Ec2Client:           ec2Client,
+		},
+		Estimator: &estimator.Estimator{
+			ScannerRegion:       config.ScannerRegion,
+			ScannerInstanceType: config.ScannerInstanceType,
+			Ec2Client:           ec2Client,
```

Comment:
I would extend the `Estimator` with pricing client attribute (e.g. `Estimator.PricingClient`) and initialize the pricing client once instead of every time when the `Estimator.Estimate()` method is called here: https://github.com/openclarity/vmclarity/pull/1324/files#diff-961df5bd86ba8aa80d2445c5b7aaba4158d2c9b3d6f853722a55bd6ceb2ee694R42-R47

Reply:
good point, yes 👍 let me update this